client-js/app/elements/labrad-plot1d.ts

```diff
@@ -252,4 +258,46 @@ export class Plot1D extends polymer.Base implements Plot {
          .attr('d', this.line);
     }
   }
+  
+  private reset() {
+    this.xScale.domain([this.limits.xMin, this.limits.xMax]);
+    this.yScale.domain([this.limits.yMin, this.limits.yMax]);
+    this.xAxis.scale(this.xScale);
+    this.yAxis.scale(this.yScale);
+    this.zoom.x(this.xScale);
+    this.zoom.y(this.yScale);
+    this.handleZoom();
+  }
+
+  private zoomRectangle() {
+    var e = this,
+        origin = d3.mouse(e),
+        rect = e.svg.append('rect')
+	    .attr('stroke', 'red')
+            .attr('fill-opacity', 0.5);
+
+    e.svg
+        .on('mousemove.zoomRect', function() {
```

Comment:
In typescript, can use arrow functions instead:
```javascript
.on('mousemove.zoomRect', () => {
 ...
});
```
One advantage is that arrow functions bind `this` with lexical scope, instead of dynamic scope as in plain javascript. This would obviate the need for the local alias `e` for `this` in the outer function. (This file generally has some bad examples of aliasing `this`, mostly for brevity. We should probably clean that up at some point...)